src/tools/blast/components/results/HSPDetailPanel.tsx

```diff
@@ -0,0 +1,383 @@
+        range.end = i;
+        ranges.push([range.start, range.end]);
+        range = newRange();
+      }
+
+      if (i === seq.length - 1 && range.end === null) {
+        range.end = i + 1;
+        ranges.push([range.start as number, range.end as number]);
+        range = newRange();
+      }
+    });
+
+    return ranges;
+  };
+
+  const setQueryTrackData = useCallback(
+    (node): void => {
+      if (node) {
+        const seqSegments = findSequenceSegments(hsp_qseq);
+        const offset = Math.abs(hsp_hit_from - hsp_query_from);
+        const opacity = hsp_identity / 100;
+
+        const blockSegments = seqSegments.map(([start, end]) => ({
+          start: start + offset,
+          end: end + offset,
+          // franklin $colour-sapphire-blue
+          color: '#014371',
+          opacity,
+        }));
+
+        const lineSegments = [
+          {
+            start: offset + 1,
+            end: hsp_query_from + 1,
+            shape: 'line',
+            color: '#014371',
+            opacity,
+          },
+          {
+            start: blockSegments[blockSegments.length - 1].end + 1,
+            end: queryLength + offset,
+            shape: 'line',
+            color: '#014371',
+            opacity,
+          },
+        ];
+
+        // eslint-disable-next-line no-param-reassign
+        node.data = [...lineSegments, ...blockSegments];
+      }
+    },
+    [hsp_query_from, hsp_query_to, hsp_identity, hitLength]
+  );
+
+  const setMatchTrackData = useCallback(
+    (node): void => {
+      if (node) {
+        const seqSegments = findSequenceSegments(hsp_hseq);
+        const offset = Math.abs(hsp_hit_from - hsp_query_from);
+        const opacity = hsp_identity / 100;
+
+        const blockSegments = seqSegments.map(([start, end]) => ({
+          start: start + offset,
+          end: end + offset,
+          // franklin $colour-sapphire-blue
+          color: '#014371',
+          opacity,
+        }));
+
+        const lineSegments = [
+          {
+            start: 1,
+            end: offset,
+            shape: 'line',
+            color: '#014371',
+            opacity,
+          },
+          {
+            start: blockSegments[blockSegments.length - 1].end + 1,
+            end: hitLength,
+            shape: 'line',
+            color: '#014371',
+            opacity,
+          },
+        ];
+
+        // eslint-disable-next-line no-param-reassign
+        node.data = [...lineSegments, ...blockSegments];
+      }
+    },
+    [hsp_hit_from, hsp_hit_to, hsp_identity, hitLength]
+  );
+
+  const setMSAAttributes = useCallback(
+    (node): void => {
+      if (!node) {
+        return;
+      }
+      node.data = [
+        {
+          name: 'Query',
+          sequence: hsp_qseq,
+        },
+        {
+          name: 'Match',
+          sequence: hsp_hseq,
+        },
+      ];
+
+      node.onActiveTrackChange = (trackId: string) => {
```

Comment:
I don't think we need this for BLAST, only for Align

Reply:
Removed